project/cmake/scripts/common/projectmacros.cmake

```diff
@@ -50,10 +50,32 @@ endfunction()
 # Copied from FindGTest.cmake 
 # Thanks to Daniel Blezek <blezek@gmail.com> for the GTEST_ADD_TESTS code
 function(GTEST_ADD_TESTS executable extra_args)
+    get_property(XBMC_GENERATED_TEST_FILES GLOBAL PROPERTY XBMC_GENERATED_TEST_FILES)
```

Comment:
Hmm, so I'm looking into this and it looks like such a change would either impact quite a lot of other code or would be kind of weird.

The first solution would be to have to have two variables, `SOURCES` and `SUPPORT_SOURCES`. Both of them are used by `add_library`, except that only the former is used by `gtest_add_tests`. The downside here is that it isn't immediately obvious that `SOURCES` is supposed to exclude "support" files.

The second solution would be to have `TEST_SOURCES` and `SUPPORT_SOURCES`, which are both merged into `SOURCES` at the time `core_add_test` is run. That would require all the tests' CMakeLists to be updated.

Since its the least invasive, I'll implement the first so we can see how it looks.